src/classes/CompensationsController.cls

```diff
@@ -2,18 +2,32 @@ public with sharing class CompensationsController {
 
 	@AuraEnabled
 	public static List<Compensation__c> getCompensations(String recordType, String contactId) {
-        System.debug(recordType);
-        System.debug(contactId);
+
+        List<Compensation__c> comps = new List<Compensation__c>();
+
+
         if (recordType == 'all'){
             if (contactId != null){
-                return [SELECT Id, Name, Contact__r.Id, Contact__r.Name, Contact__r.Birthdate, Job_Category__c, RecordType.Name, Amount__c, Location__c, Office__c, Submitted__c, Status__c FROM Compensation__c WHERE Contact__r.Id =:contactId];
+                for (List<Compensation__c> com : [SELECT Id, Name, Contact__r.Id, Contact__r.Name, Contact__r.Birthdate, Job_Category__c, RecordType.Name, Amount__c, Location__c, Office__c, Submitted__c, Status__c FROM Compensation__c WHERE Contact__r.Id =:contactId]) {
```

Comment:
This is better, but still needs a LIMIT statement to prevent the user from reaching the governor limit even though it can be higher this way... just LIMIT 50000 is fine.